### DISCUSSION TITLE
'register' not working with 'shouldUnregister: true'

### DISCUSSION THREAD

**Discussion by alexanderleonenko** [react-hook-form/react-hook-form#5631] [Q&A] [answered]
**Describe the bug**
I'm trying to use react-hook-form V7 with `shouldUnregister: true` and keep field value by using `form.register('field')`

**To Reproduce**

1. Go to [codesandbox](https://codesandbox.io/s/elastic-resonance-j1pmq?file=/src/App.js)
2. Click on 'toggle field' button to hide `Controller`
2. Click on 'toggle field' button again to show `Controller`

**Codesandbox link (Required)**
[https://codesandbox.io/s/elastic-resonance-j1pmq?file=/src/App.js](https://codesandbox.io/s/elastic-resonance-j1pmq?file=/src/App.js)

**Expected behavior**
Keep `field: "field"` value by using `form.register("field")`

**Screenshots**
![capture](https://user-images.githubusercontent.com/85992052/122194653-edccc100-ce95-11eb-9650-d8f6d5cab000.gif)

**Comment by bluebill1049** [comment, marked as answer]
what do you want to achieve with the above? you unmount an input but then you are still calling `register` before render?

**Reply by alexanderleonenko** [reply]
Like in V6 I want to have `shouldUnregister: true` for most fields and at the same time store some fields' values without `Controller`

**Reply by bluebill1049** [reply]
`register('test', { value: 'test' })`

Is this what you after? 

also if you want to store that single input, you probably want to do in `useEffect` right with the state as dep?

**Reply by alexanderleonenko** [reply]
`useEffect()` helped. Thanks
